REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/user/sevice/UserServiceImpl.java

```diff
@@ -51,4 +53,37 @@ private UserNotFoundException userNotFoundException() {
         log.error(ErrorConstants.USER_NOT_FOUND_MESSAGE);
         return new UserNotFoundException();
     }
+
+    @Override
+    public void createUser(UserDetails user) {
+        ((User) user).setPassword(passwordEncoder.encode(user.getPassword()));
+        userRepository.save((User) user);
+    }
```

Comment:
Can't you make this method input parameter User?

Reply:
thats method from UserDetailsManager interface.
After a little thought, I came to the conclusion that it was not needed, so I removed it